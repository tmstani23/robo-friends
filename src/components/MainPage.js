import React from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundry';
import Header from '../components/Header';
//import '../styles/fonts/App.css';



class MainPage extends React.Component {

    componentDidMount() {
        this.props.onRequestRobots()
    }

    filterRobots = () => {
        //Filter the robots to only include those in the searchfield
        return this.props.robots.filter(robot => {   
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
        })
    }

    render() {
        
        //searchField and searchChange now coming from redux store as props
        const {onSearchChange, isPending} = this.props;
        
        
            

        return (
            <div className='tc'>
                <Header />
                
                <Scroll>
                    { isPending ? <h1>Loading...</h1>
                        : <ErrorBoundary>
                            <SearchBox searchChange={onSearchChange}/>
                            <CardList robots={this.filterRobots()}/> 
                        </ErrorBoundary>
                    }
                </Scroll>
                
            </div>
        )  
    }
    
}

export default MainPage;