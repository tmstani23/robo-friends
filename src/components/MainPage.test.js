import {shallow} from 'enzyme';
import React from 'react';
import MainPage from './MainPage';
import renderer from 'react-test-renderer';

let wrapper;
//jest method that runs before each test
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps} />)
})

it('expect app component to render pending false', () => {
    
    const tree = renderer.create(
        wrapper
      ).toJSON();
      expect(tree).toMatchSnapshot();
})

it('expect app component to render div when pending true', () => {
    const mockPropsPending = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: true
    }
    let wrapperPending = shallow(<MainPage {...mockPropsPending} />)
    const treePending = renderer.create(
        wrapperPending
      ).toJSON();
      expect(treePending);
      expect(treePending.children[0].type).toEqual("div")
})

it('filters the robots correctly2', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id:3,
            name:'Johnny',
            email:'Johnny@gmail.com'
        }],
        searchField: 'Johnny',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2} />)
    expect(wrapper2.instance().filterRobots()).toEqual(mockProps2.robots);
})

it('searchbox and filter working', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id:3,
            name:'Johnny',
            email:'Johnny@gmail.com'
        }],
        searchField: 'a',
        isPending: false
    }
    const wrapper3 = shallow(<MainPage {...mockProps3} />)
    const filteredRobots = [];

    expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
})