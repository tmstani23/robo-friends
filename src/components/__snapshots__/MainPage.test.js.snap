// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect app component to render pending false 1`] = `
<div
  className="tc"
>
  <div>
    <button
      color="red"
      onClick={[Function]}
    >
      Count: 
      0
    </button>
    <h1
      className="f1"
    >
      RoboFriends
    </h1>
  </div>
  <div
    style={
      Object {
        "border": "1px solid #0ccac4",
        "height": "500px",
        "overflow": "scroll",
      }
    }
  >
    <div
      className="pa2"
    >
      <input
        aria-label="Search Robots"
        className="pa3 ba b--green bg-lightest-blue"
        placeholder="search robots"
        type="search"
      />
    </div>
    <h1
      className="tc"
    >
       Loading...
    </h1>
  </div>
</div>
`;
